astropy/table/pprint.py

```diff
@@ -394,7 +394,7 @@ def _pformat_col_iter(self, col, max_lines, show_name, show_unit, outs,
             try:
                 dtype = dtype_info_name(col.dtype)
             except AttributeError:
-                dtype = 'object'
+                dtype = col.__class__.__qualname__ or 'object'
```

Comment:
I'm feeling deja vu about this question, but is there any way that a class can have a `__qualname__` that is empty (evals to False)? I would just skip the `or 'object'`.

Reply:
Definitely not common, but a dynamically created class can have an empty qualname. I'm trying to recall, but I think I put in the ``or 'object'``  because of this scenario.

```python
A = type("", (object,), {})
A.__qualname__
```

Reply:
I see, OK.